File: 1677-matrix-diagonal-sum/1677-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        total = 0
        row = len(mat[0])
        for i in range(row):
            total += mat[i][i]
            total += mat[i][row - 1 - i]
        if len(mat) % 2 != 0:
            total -= mat[row // 2][row // 2]
        return total

File: 1677-matrix-diagonal-sum/1677-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum = 0;
        int row = mat[0].length;
        for(int i = 0; i < row; i++){
            for(int j = 0; j < row; j++){
                if(i == j){
                    sum += mat[i][j];
                }
            }
            sum += mat[i][row - i - 1];
        }
        if(row % 2 != 0){
            sum -= mat[row / 2][row / 2];
        }
        return sum;
    }
}